### PR TITLE
[release/v2.21] Add keepalive-time setting for Konnectivity

### PR DESCRIPTION
**Pull request by mfranczy**
**What this PR does / why we need it**:
Manual backport of https://github.com/kubermatic/kubermatic/pull/11502

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add keepalive-time Konnectivity setting + set keepalive to 1m by default
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by embik** [comment]
Hey @mfranczy, this is still missing approval and lgtm labels. Release managers will approve it afterwards :-)

**Comment by mfranczy** [comment]
@ahmedwaleedmalik @moadqassem or @xmudrii could you take a look?

**Comment by xmudrii** [comment]
@mfranczy Can you please get this PR approved first? After that, we can cherry-pick-approve it.

File: cmd/user-cluster-controller-manager/main.go
Comment (xmudrii) [inline cmd/user-cluster-controller-manager/main.go:104]: Nit: why is this not a duration type, e.g. `time.Time` or `metav1.Time`?

**Review by xmudrii — APPROVED** [review]
I have one nit, but that anyways has to be fixed on main first (if we decide to fix it), so I'm not going to block on it.
/lgtm
/approve

File: cmd/user-cluster-controller-manager/main.go
Comment (mfranczy) [reply cmd/user-cluster-controller-manager/main.go:104]: Because Konnectivity's `keepalive-time` option is a `flag.DurationVar`  .
`ParseDuration`  expect string as input, that's why it is a string as we just proxy the parameter to konnectivity.

https://pkg.go.dev/time#ParseDuration

**Pull request merged**
